src/ceph-disk/ceph_disk/main.py

```diff
@@ -1666,26 +1666,33 @@ def create_partition(self, uuid, name, size=0, num=0):
         if num == 0:
             num = get_free_partition_index(dev=self.path)
         if size > 0:
-            new = '--new={num}:0:+{size}M'.format(num=num, size=size)
+            end = '{size}M'.format(size=size)
             if size > self.get_dev_size():
                 LOG.error('refusing to create %s on %s' % (name, self.path))
                 LOG.error('%s size (%sM) is bigger than device (%sM)'
                           % (name, size, self.get_dev_size()))
                 raise Error('%s device size (%sM) is not big enough for %s'
                             % (self.path, self.get_dev_size(), name))
         else:
-            new = '--largest-new={num}'.format(num=num)
+            end = '100%'
 
         LOG.debug('Creating %s partition num %d size %d on %s',
                   name, num, size, self.path)
         command_check_call(
             [
+               'parted',
```

Comment:
I researched the sgdisk and parted, seems I made a mistake, sgdisk also uses GPT which supports 2TB parted. From man page of sgdisk, sgdisk has some limitation on MBR conversion and BSD partition, but it's not strong reason to switch it.

I picked the trello item "switch sgdisk to parted", but now I also have no idea why shall we switch to parted. How can we know the original purpose of this backlog item? If can't know, what do you think? Should I drop this item?